python/lsst/qa/explorer/dataset.py

```diff
@@ -1,806 +1,329 @@
-            df_dict = {k:df_swap[k].dropna(how='any', subset=subset_to_check).reset_index() 
-                            for k in ['coadd'] + self.catalog.visit_names}
-            self._ds_dict = {k:hv.Dataset(df_dict[k], kdims=kdims, vdims=vdims) for k in df_dict}
-
-            # Keep only rows that aren't nan in visit values
-            df = pd.concat(dfs, axis=1, join='inner')
-        else:
-            df = self.df.dropna(how='any')
-
-        ds = hv.Dataset(df.reset_index(), kdims=kdims, vdims=vdims)
-        self._ds = ds        
-
-    def color_ds(self, mag):
-        """Calculate holoviews.Dataset object containing colors for a given magnitude type
-
-        * Functor values and 'x' values come from catalog's reference filter.
-        * Flags are computed as the "or" of all bands.
-        * Labels are re-determined as follows: 
-            * If object is a star in all bands, it is called a "star"
-            * If it is a star in zero bands, it is called a "noStar"
-            * If it is called a star in some bands but not all, it is called a "maybe"
-        """
-        if not self.is_multiband:
-            return NotImplementedError('Can only get color_ds if catalog is a MultiBandCatalog')
-        if not isinstance(self.allfuncs[mag], Mag):
-            raise ValueError('Requested column must be a magnitude: {} requested'.format(mag))
-
-        color_df = self.df[['ra', 'dec']]
-        color_df.columns = color_df.columns.get_level_values(0)
-
-        filt = self.catalog.reference_filt
-        swap_df = self.df.swaplevel(axis=1)
-
-        # Get values for functors and 'x' from reference filter
-        func_keys = list(self.funcs.keys()) + ['x'] + [self.id_name]
-        color_df = pd.concat([color_df, swap_df[filt][func_keys]], axis=1)
-
-        # Compute flags as the "or" of all 
-        flag_cols = [pd.Series(self.df[flag].max(axis=1).astype(bool), name=flag) for flag in self.flags]
-        color_df = pd.concat([color_df] + flag_cols, axis=1)
-
-        # Calculate group label
-        n = self.catalog.n_filters
-        n_star = (self.df['label']=='star').sum(axis=1)
-        label = pd.Series(pd.cut(n_star, [-1, 0, n-1 , n], labels=['noStar', 'maybe', 'star']),
-                                    index=n_star.index, name='label')
-        color_df['label'] = label
-
-        color_df = pd.concat([color_df, self.df['{}_color'.format(mag)]], axis=1)
-
-        # color_df = pd.concat([self.df[['ra', 'dec']], 
-        #                       swap_df[filt], 
-        #                       self.df['{}_color'.format(mag)]], axis=1)
-        # color_df = color_df.rename(columns={('ra', 'ra'):'ra', ('dec', 'dec'): 'dec'})    
-
-        return hv.Dataset(color_df, kdims=self._get_kdims())
+    def _makeDataset(self):
+        kdims, vdims = self._getDims()
 
+        df = self.df.replace([np.inf, -np.inf], np.nan).dropna(how='any')
```

Comment:
This is to facilitate plotting with holoviews; at least when I was developing this, infs or nans would cause the plots to barf, so I get rid of them.  The `hv.Dataset` object created here (`self.ds`) is only the support vehicle for the plots, and the underlying `self.df` is unaffected by this.  Does this also answer your above question about infs/nans?